### PR TITLE
(#1008) Fixed HttpServletResponseFake implementation

### PR DESCRIPTION
**Pull request by fanifieiev**
This PR is for #1008

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#1023](https://github.com/yegor256/takes/issues/1023) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/13560fa9-c28c-4a48-81f0-ad4b743960e9); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by paulodamaso** [comment]
@0crat assign me

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/takes/issues/1023#issuecomment-581579691))

@paulodamaso @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) not enough funds available in the project, can't set budget of job [#1023](https://github.com/yegor256/takes/issues/1023), see [§21](http://www.zerocracy.com/policy.html#21); @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) will get no money on completion; in order to fix that, add funds to the project and assign the job again

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/takes/issues/1023#issuecomment-581579691))

@paulodamaso There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@377a78bde909, thread: PQ-C3FFK3YAY
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [comment, job, issue, repo, cause, _parent_sig, flow, login]
```

[0.51.0](https://github.com/zerocracy/farm): CID: [7ec1d5df-455e-4ce6-b28a-0569727f48f3](https://www.0crat.com/footprint/C3FFK3YAY/7ec1d5df-455e-4ce6-b28a-0569727f48f3), Type: "Order was canceled"

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public void cookie() throws Exception {
         sresp.addCookie(new Cookie(name, value));
         MatcherAssert.assertThat(
             "Can't add a cookie in servlet response",
-            sresp.getHeaders(name),
+            sresp.getHeaders("Set-Cookie:"),
```
Comment (paulodamaso) [inline src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @fanifieiev Let's put this in a constant.

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 2]
```diff
@@ -113,7 +113,7 @@ public void status() {
         sresp.setStatus(502);
         MatcherAssert.assertThat(
             "Can't set a status in servlet response",
-            sresp.getHeaders("Status"),
+            sresp.getHeaders("HTTP/1.1"),
```
Comment (paulodamaso) [inline src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @fanifieiev Let's put this in a constant.

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 3]
```diff
@@ -127,7 +138,7 @@ public void sendError() throws IOException {
         sresp.sendError(101, "Custom error message");
         MatcherAssert.assertThat(
             "Can't send a error in servlet response",
-            sresp.getHeaders("Status"),
+            sresp.getHeaders("HTTP/1.1"),
```
Comment (paulodamaso) [inline src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @fanifieiev Why not use the constant here too?

Comment (fanifieiev) [reply src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @paulodamaso Missed that, will fix.

**Comment by fanifieiev** [comment]
@paulodamaso Please review again

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 4]
```diff
@@ -41,6 +41,27 @@
...
+
+    /**
+     * HTTP/1.1 502 bad gateway.
+     */
+    private static final String HTTP_1_1_502 = "HTTP/1.1 502 Bad Gateway";
+
+    /**
+     * HTTP/1.1 101 custom error message.
+     */
+    private static final String HTTP_1_1_101 =
```
Comment (paulodamaso) [inline src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @fanifieiev And here you could call thsi constant `INFO`. Also, the code and message does not relate each other: HTTP error code 101 reason phrase is `Switching Protocols`.

File: src/main/java/org/takes/servlet/HttpServletResponseFake.java [hunk 5]
```diff
@@ -123,18 +117,13 @@ public void sendError(
...
         try {
-            return new ListOf<>(
-                new Filtered<>(
-                    hdr -> !new UncheckedText(
-                        new Lowered(hdr)
-                    ).asString().startsWith(prefix),
-                    this.response.get().head()
-                )
+            return new Filtered<>(
+                hdr -> new Lowered(hdr).asString().startsWith(prefix),
```
[the remark this answers is not in the document]
Comment (fanifieiev) [reply src/main/java/org/takes/servlet/HttpServletResponseFake.java]: @paulodamaso I wish I could use it, but it is not available  in Takes.
Still waiting for https://github.com/yegor256/takes/issues/1025 to be done.

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 6]
```diff
@@ -113,8 +135,10 @@ public void status() {
         sresp.setStatus(502);
         MatcherAssert.assertThat(
             "Can't set a status in servlet response",
-            sresp.getHeaders("Status"),
-            Matchers.hasItem("HTTP/1.1 502 Bad Gateway")
+            sresp.getHeaders(HttpServletResponseFakeTest.HTTP_1_1),
+            Matchers.hasItem(
```
[the remark this answers is not in the document]
Comment (fanifieiev) [reply src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @paulodamaso I have not added that, just formatted the code, but will fix that as well

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 7]
```diff
@@ -127,8 +151,10 @@ public void sendError() throws IOException {
         sresp.sendError(101, "Custom error message");
         MatcherAssert.assertThat(
             "Can't send a error in servlet response",
-            sresp.getHeaders("Status"),
-            Matchers.hasItem("HTTP/1.1 101 Custom error message")
+            sresp.getHeaders(HttpServletResponseFakeTest.HTTP_1_1),
+            Matchers.hasItem(
```
[the remark this answers is not in the document]
Comment (fanifieiev) [reply src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @paulodamaso I have not added that, just formatted the code, but will fix that as well

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java (hunk 4 above)
Comment (fanifieiev) [reply src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @paulodamaso OK

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 8]
```diff
@@ -41,6 +41,27 @@
...
+
+    /**
+     * HTTP/1.1 header name.
+     */
+    private static final String HTTP_1_1 = "HTTP/1.1";
+
+    /**
+     * HTTP/1.1 502 bad gateway.
+     */
+    private static final String HTTP_1_1_502 = "HTTP/1.1 502 Bad Gateway";
```
[the remark this answers is not in the document]
Comment (fanifieiev) [reply src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @paulodamaso OK

File: src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java [hunk 9]
```diff
@@ -41,6 +41,27 @@
...
 public final class HttpServletResponseFakeTest {
+    /**
+     * Set-Cookie header name.
+     */
+    private static final String SET_COOKIE = "Set-Cookie:";
+
+    /**
+     * HTTP/1.1 header name.
+     */
+    private static final String HTTP_1_1 = "HTTP/1.1";
```
[the remark this answers is not in the document]
Comment (fanifieiev) [reply src/test/java/org/takes/servlet/HttpServletResponseFakeTest.java]: @paulodamaso OK

**Comment by fanifieiev** [comment]
@paulodamaso Please have a look one more time.

File: src/main/java/org/takes/servlet/HttpServletResponseFake.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (paulodamaso) [reply src/main/java/org/takes/servlet/HttpServletResponseFake.java]: @fanifieiev OK, let's wait then

**Comment by paulodamaso** [comment]
@fanifieiev I'ts good now, thanks!

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor stop

**Comment by 0crat** [comment]
Job [#1023](https://github.com/yegor256/takes/issues/1023) is not in the agenda of @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), can't retrieve data and time of add

**Comment by 0crat** [comment]
Code review was too long (9 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Pull request merged**